### PR TITLE
Use can-globals/document/document to get the global document

### PR DESCRIPTION
**Pull request by matthewp**
Rather than checking if a node is inserted into its ownerDocument, check
if it is inserted into the global document. This means you can create
nodes from other documents and then insert them into the global document
and have onNodeInsertion work. This is what happens in ssr.

### CODE REVIEW

File: test/can-dom-mutate-test.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ var test = unit.test;
 var moduleMutationObserver = testUtils.moduleMutationObserver;
 
 moduleMutationObserver('can-dom-mutate', function () {
-	test('onNodeInsertion should be called when that node is inserted', function (assert) {
+	test('onNodeInsertion  ', function (assert) {
```
Comment (andrejewski) [inline test/can-dom-mutate-test.js]: Undo this change, and everything should be good. 👍

Comment (matthewp) [reply test/can-dom-mutate-test.js]: oops! thanks for catching.
